web/projects/ui/src/app/pages/apps-routes/app-interfaces/app-interfaces.page.ts

```diff
@@ -126,3 +100,65 @@ export class AppInterfacesItemComponent {
     await toast.present()
   }
 }
+
+function getAddresses(
+  serviceInterface: ServiceInterfaceWithHostInfo,
+): MappedAddress[] {
+  const host = serviceInterface.hostInfo
+  const addressInfo = serviceInterface.addressInfo
+  const username = addressInfo.username ? addressInfo.username + '@' : ''
+  const suffix = addressInfo.suffix ? '/' + addressInfo.suffix : ''
+
+  const hostnames =
+    host.kind === 'multi'
+      ? host.hostnames
+      : host.hostname
+      ? [host.hostname]
+      : []
+
+  return hostnames
+    .map(h => {
+      const addresses: MappedAddress[] = []
+
+      let name = ''
+      let hostname = ''
+
+      if (h.kind === 'onion') {
+        name = 'Tor'
+        hostname = h.hostname.value
+      } else {
+        name = h.hostname.kind
+        hostname =
+          h.hostname.kind === 'domain'
+            ? `${h.hostname.subdomain}.${h.hostname.domain}`
+            : h.hostname.value
+      }
+
+      if (h.hostname.sslPort) {
+        const port = h.hostname.sslPort === 443 ? '' : `:${h.hostname.sslPort}`
```

Comment:
Don't hardcode 443. Only erase the port if it matches the defaultPort for a KnownProtocol. So look up the scheme in knownProtocols, and check the port against KnownProtocol.defaultPort

Reply:
These are always UIs

Reply:
Then I suppose it technically would work, though I'd document the reliance on all uis being http. Personally I'd abstract url construction to a different function and handle it there though cause you'll need it for displaying interface urls too